fenix/app/src/main/java/org/mozilla/fenix/debugsettings/store/DebugDrawerNavigationMiddleware.kt

```diff
@@ -5,29 +5,41 @@
 package org.mozilla.fenix.debugsettings.store
 
 import androidx.navigation.NavHostController
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 import mozilla.components.lib.state.Middleware
 import mozilla.components.lib.state.MiddlewareContext
 import org.mozilla.fenix.debugsettings.navigation.DebugDrawerRoute
+import org.mozilla.fenix.debugsettings.ui.DEBUG_DRAWER_HOME_ROUTE
 
 /**
  * Middleware that handles navigation events for the Debug Drawer feature.
  *
  * @param navController [NavHostController] used to execute any navigation actions on the UI.
+ * @param scope [CoroutineScope] used to make calls to the main thread.
  */
 class DebugDrawerNavigationMiddleware(
     private val navController: NavHostController,
+    private val scope: CoroutineScope,
 ) : Middleware<DebugDrawerState, DebugDrawerAction> {
 
     override fun invoke(
         context: MiddlewareContext<DebugDrawerState, DebugDrawerAction>,
         next: (DebugDrawerAction) -> Unit,
         action: DebugDrawerAction,
     ) {
-        when (action) {
-            is DebugDrawerAction.NavigateTo.Home -> navController.navigate(route = DebugDrawerRoute.Home.route)
-            is DebugDrawerAction.NavigateTo.TabTools -> navController.navigate(route = DebugDrawerRoute.TabTools.route)
-            is DebugDrawerAction.OnBackPressed -> navController.popBackStack()
-            is DebugDrawerAction.DrawerOpened, DebugDrawerAction.DrawerClosed -> {} // no-op
+        next(action)
+        scope.launch {
+            when (action) {
+                is DebugDrawerAction.NavigateTo.Home -> navController.popBackStack(
+                    route = DEBUG_DRAWER_HOME_ROUTE,
+                    inclusive = false,
+                )
```

Comment:
The previous implementation would have just added Home to the back stack, rather than taking the user back to the root. This change will make it so any future call with `DebugDrawerAction.NavigateTo.Home` will properly send the user to the root and clear the back stack.